builtin/providers/aws/structure.go

```diff
@@ -1276,3 +1277,139 @@ func flattenApiGatewayThrottleSettings(settings *apigateway.ThrottleSettings) []
 
 	return result
 }
+
+// TODO: refactor some of these helper functions and types in the terraform/helper packages
```

Comment:
Nice! With a few unit tests these helpers could do a lot of good for the community. 👍